final_task/rss_reader/output_functions.py

```diff
@@ -0,0 +1,319 @@
+        pass
+    logging.info("Got some alternative text.")
+    return pack_of_alts
+
+
+def getting_pack_of_news(the_feed, main_source, num_of_news=None):
+    """
+    Creating full novelty
+    1. Adding list of images, correct it if some duplicates are there
+    2. Adding list of alternative texts, correct it if some duplicates are there
+    3. Some problems can occur if there are no alternative text, solving them with changing list of alts
+    """
+    if getting_num_of_news(the_feed, num_of_news) > len(the_feed.entries):
+        print("You want to get more news than we can get!")
+        print("Printing all news from the storage!")
+
+    pack_of_news = []
+    pack_of_news_for_db = []
+    pack_of_images_links = getting_images_links(the_feed, num_of_news)
+    pack_of_alts = getting_alt_text(the_feed, num_of_news)
+    corrected_pack_of_alts = [el for el, _ in groupby(pack_of_alts)]
+
+    for num, item in enumerate(the_feed.entries[:num_of_news]):
+        novelty, novelty_for_database = getting_novelty(item, num, pack_of_images_links, corrected_pack_of_alts,
+                                                        main_source)
+        pack_of_news.append(novelty)
+        pack_of_news_for_db.append(novelty_for_database)
+    return pack_of_news, pack_of_news_for_db
+
+
+def getting_novelty(item, number, corrected_pack_of_images_links, corrected_pack_of_alts, main_source):
+    """
+    It was really difficult to get images because of variety of templates how sites leave link for image
+    Here we create object of Novelty class and fill it with our title, description and etc.
+    Then if some problems with images or alt.text occur we use
+    """
+
+    try:
+        alt_text = corrected_pack_of_alts[number]
+    except IndexError:
+        alt_text = 'No alternative text.'
+    novelty = Novelty(number + 1, unescape(item.get('title', '')),
+                      item.get('published', ''),
+                      item.get('link', ''),
+                      unescape(clean_html(item.get('description', ''))),
+                      corrected_pack_of_images_links[number],
+                      unescape(alt_text),
+                      getting_corrected_time(item),
+                      main_source)
+    novelty_for_database = (novelty.number_of_novelty, novelty.title_of_novelty, novelty.time_of_novelty,
+                            novelty.source_link, novelty.description, novelty.images_links,
+                            novelty.alt_text, novelty.date_corrected, novelty.main_source)
+    return novelty, novelty_for_database
+
+
+def getting_full_info(the_feed, pack_of_news):
+    """
+    Getting full info from news
+    try-except for printing links and alternative text
+    """
+    print("------------------------")
+    print("Source: ", printing_title(the_feed))
+    for novelty in pack_of_news:
+        print()
+        print("{0}.".format(novelty.number_of_novelty), "Title: ", novelty.title_of_novelty)
```

Comment:
Here and everywhere else: if is better to use `formatted string` (`f"..."`) instead of `.format()`